wgpu-core/src/device/resource.rs

```diff
@@ -2713,7 +2713,16 @@ impl<A: HalApi> Device<A> {
             });
 
             for attribute in vb_state.attributes.iter() {
-                if attribute.offset >= 0x10000000 {
+                let format_size = attribute.format.size();
+                let attrib_end_offset = attribute.offset + format_size;
+                let offset_limit = if vb_state.array_stride == 0 {
+                    max_array_stride as u64
+                } else {
+                    vb_state.array_stride
+                }
+                .min(0x10000000);
```

Comment:
What does this magic number signify?

Reply:
I don't know, it was already there

Reply:
That's a spooky kind of magic number

Reply:
If it helps, this magic number probably first appeared in #706, where it replaced another magic number that probably first appeared in #175.